File: src/scripts/processing-data.py
#/bin/python

import csv
import sys


results = {}

def main(argv):

	path_to_results=argv[0] + "/" + argv[1] + "-0/"
	source= argv[2]
	nr_nodes = int(argv[3])

	nodes = map(lambda x: 'hostR' + str(x),  range(nr_nodes))
	for n in nodes:
		results[n] = {}

	with open(path_to_results + source + '-msg_sent', 'rb') as csvfile:
		spamreader = csv.reader(csvfile, delimiter=' ', quotechar='|')
		i = 0
		for row in spamreader:
			m_id = int(row[2])
			m_time = float(row[1])
			print m_id, m_time
			for n in nodes:
				with open(path_to_results + n + '-broadcast_msg_received', 'rb') as received:
					r2 = csv.reader(received, delimiter=' ', quotechar='|')
					for msg in r2:
						if int(msg[2]) == m_id:
							results[n][m_id] = float(msg[1]) - m_time
							break
				

	print "\n"

if __name__ == "__main__":
   main(sys.argv[1:])

File: src/scripts/processing-data.R

# ============================================
# Broadcast Time
# For each message sent from the source, we need to figure out how long it took to
# receive the message for the first time in all nodes (or all nodes that actually received)
#
# Steps:
#   1. Identify source of messages (node name)
#   2. Identify set of message sent (a table with two columns: message id, sent time)
#   3. For m in "message sent" 
#   3.    For n in nodes do
#   4.       messages-received = load messages received in that node (broadcast_msg_received)
#   5.       time = pick msg.time from message-received such that msg.id = m.id and (msg.time <= msg2.time for all msg2 in messages-received)
#   6.       results[n][m.id] = m.time - time
#
# Result
# A 2d matrix indexed by node and message id
#         m0   m1   m2
# node0  1.4  2.1  2.3
# node1  1.7  1.8  2.5
# node2  1.8  1.7  1.5
#
#

path_to_results <- "~/work/src/infocom2017/experiments/configs/results/nodes200-abba2-300broadcasts-highdensity-0/"
source <- "hostR0"

p <- paste(path_to_results, source, "-msg_sent", sep = "", collapse = "")

messages_sent <- read.table(p, quote = "\"", comment.char = "")

battery_level <- read.table(paste(path_to_results, source,"-power_level", sep=""))

printf <- function(...) invisible(cat(sprintf(...)))

sss <- apply(messages_sent, c(1,2), sum)
print(sss)

# ============================================
# Broadcast Number of Duplicated Messages
# For each message sent from the source, we need to figure out how many times a message
# is received
#
# Steps:
#   1. Identify source of messages (node name)
#   2. Identify set of message sent (a table with two columns: message id, sent time)
#   3. For m in "message sent" 
#   3.    For n in nodes do
#   4.       messages-received = load messages received in that node (broadcast_msg_received)
#   5.       n = count msg in messages_received such that m.id == msg.id 
#   6.       results[n][m.id] = n
#
# Result
# A 2d matrix indexed by node and message id
#         m0   m1   m2
# node0  1.4  2.1  2.3
# node1  1.7  1.8  2.5
# node2  1.8  1.7  1.5
#
#


# ============================================
# Battery Level for a node
# For each message sent from the source, we need to figure out how long it took to
# receive the message for the first time in all nodes (or all nodes that actually received)
#
# Steps:
#   1. Identify source of messages (node name)
#   2. Identify set of message sent (a table with two columns: message id, sent time)
#   3. For m in "message sent" 
#   3.    For n in nodes do
#   4.       messages-received = load messages received in that node (broadcast_msg_received)
#   5.       time = pick msg.time from message-received such that msg.id = m.id and (msg.time <= msg2.time for all msg2 in messages-received)
#   6.       results[n][m.id] = m.time - time
#
# Result
# A 2d matrix indexed by node and message id
#         node0
# node0  1.4  2.1  2.3
# node1  1.7  1.8  2.5
# node2  1.8  1.7  1.5
#
#
